### ISSUE TITLE
[Talos] Wrong providerID after node provisioning

### ISSUE THREAD

**Issue by ricottatosta** [ionos-cloud/cluster-api-provider-proxmox#342]
**What steps did you take and what happened:**
I have been using the following components:
- cluster-api:v1.7.2
- talos-boot:v0.6.5
- talos-cp:v0.5.6
- ipam-in-cluster:v0.1.0
- talos-ccm:v1.6.0 (running in managed cluster)

Untill capmox v0.3.0, talos-ccm has been able to set the right providerID to a new node, eg. proxmox://[unique_id].
But as of capmox v0.4.0, witih every release of capmox (I tested them all), providerID gets set to talos://nocloud/[node_ip].
Is there any brand new mechanism doing the same thing before talos-ccm? Or does talos-ccm do it wrong?

**What did you expect to happen:**
I would like talos-ccm to be able to set the right providerID.

**Anything else you would like to add:**
I am pretty sure talos-ccm sets node's providerID, because before I started using it, no providerID got set.
Am I supposed to no longer use talos-ccm?

**Environment:**

- Cluster-api-provider-proxmox version: v0.4.0 and later
- Kubernetes version: (use `kubectl version`): v1.28.3
- OS (e.g. from `/etc/os-release`): Talos v1.6.8

**Comment by mcbenjemaa** [comment]
Capmox now also has `providerId` in the metadata.
[provider-id](https://github.com/ionos-cloud/cluster-api-provider-proxmox/blob/51a9d0a454937e3a12f74cbdf287afe10756e1b7/pkg/cloudinit/metadata.go#L23C1-L23C12)

@ekarlso used that to initialize the nodes

**Comment by ricottatosta** [comment]
Yes, I know about that. But I don't know neither @ekarlso setup nor if next releases will be effective against my setup.
And that's for the future.
For the past, all my own deployments depends on a capmox controller I modify and rebuild. Untill release 0.3.0 my own changes work well with Talos, but since 0.4.0 it's no longer the same. I would like to know what changes in capmox releases later than 0.3.0 have been affecting mine.
It would be a pity if I had to give up using Talos cloud controller manager as it does more than setting a providerId.

**Comment by wikkyk** [comment]
#347 makes provider-id opt-in.

**Comment by mcbenjemaa** [comment]
Please check with the new release v0.6.0

**Comment by wikkyk** [comment]
Actually, please check with v0.6.1 :)

**Comment by wikkyk** [comment]
I believe this is fixed now. Feel free to reopen if not.
